src/Endpoints/class-recommendations-api-proxy.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ * Parsely Recommendations REST API Endpoint
+ *
+ * @package Parsely
+ */
+
+namespace Parsely\Endpoints;
+
+use Parsely\Recommended_Content;
+use stdClass;
+use WP_Error;
+use WP_REST_Server;
+use WP_REST_Request;
+
+/**
+ * A "namespace" class with functions that power REST API endpoints for use by the Recommendations Block and other consumers.
+ */
+final class Recommendations_API_Proxy {
+	/**
+	 * Entrypoint to register the endpoint and otherwise initialize this class.
+	 *
+	 * @return void
+	 */
+	public function run() {
+		self::register_rest_route();
+	}
+
+	/**
+	 * Registers the REST API Routes and backing functionality
+	 *
+	 * @return void
+	 */
+	public static function register_rest_route() {
+		register_rest_route(
+			'wp-parsely/v1',
+			'/recommendations',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => __CLASS__ . '::get_items',
+					'permission_callback' => __CLASS__ . '::permission_callback',
+					'args'                => array(),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Determine if the endpoint can be called. Applies `read` permissions.
+	 *
+	 * @return boolean
+	 */
+	public static function permission_callback() {
+		// Unauthenticated.
+		return true;
+	}
+
+	/**
+	 * Cached "proxy" to the Parsely `/related` endpoint
+	 *
+	 * @param WP_Rest_Request $request The request object.
+	 * @return array
+	 */
+	public static function get_items( WP_Rest_Request $request ) {
+		global $parsely;
+		$options = $parsely->get_options();
+		$apikey  = $options['apikey'];
+		$params  = $request->get_params();
+
+		if ( empty( $apikey ) ) {
+			return (object) array(
+				'data'  => array(),
+				'error' => new WP_Error( 400, __( 'A Parsely API Key must be set in site options to use this Endpoint', 'wp-parsely' ) ),
+			);
+		}
+
+		$cache_key = 'api_recos-' . md5( wp_json_encode( compact( 'apikey', 'params' ) ) );
```

Comment:
sonarcloud is complaining that md5 is a security risk...  Should I replace this with a more robust hashing algorithm?
```suggestion
		$cache_key = 'api_recos-' . hash( 'sha512', wp_json_encode( compact( 'apikey', 'params' ) ) );
```

Reply:
`wp_hash()`?

Reply:
+1 on using `wp_hash` and a modern hashing algorithm.